### ISSUE TITLE
'arithmetic operation overflowed' when reading "-0" as JSON

### ISSUE THREAD

**Issue by tonyg** [serde-rs/json#36]
Running this program:

```rust
extern crate serde_json;
fn main() {
    serde_json::from_str::<serde_json::Value>("-0").unwrap();
}
```

causes:

```
thread '<main>' panicked at 'arithmetic operation overflowed', [...]/.cargo/registry/src/github.com-88ac128001ac3a9a/serde_json-0.6.0/src/de.rs:289
```

whereas I expected it to run to completion successfully with no output.

**Comment by nixpulvis** [comment]
See #35

**Comment by erickt** [comment]
This was closed in #35.
